src/components/leaf.js

```diff
@@ -88,23 +88,16 @@ class Leaf extends React.Component {
     const text = this.renderText(props)
     if (el.textContent != text) return true
 
-    // If the selection was previously focused, and now it isn't, re-render so
-    // that the selection will be properly removed.
-    if (this.props.state.isFocused && props.state.isBlurred) {
-      const { index, node, ranges, state } = this.props
-      const { start, end } = OffsetKey.findBounds(index, ranges)
-      if (state.selection.hasEdgeBetween(node, start, end)) return true
-    }
-
-    // If the selection will be focused, only re-render if this leaf contains
-    // one or both of the selection's edges.
-    if (props.state.isFocused) {
+    // If the selection will be focused and native selection is not already
+    // in syncro with it, update native selection.
+    if (props.state.isFocused && props.selectionNeedsRedraw) {
       const { index, node, ranges, state } = props
       const { start, end } = OffsetKey.findBounds(index, ranges)
-      if (state.selection.hasEdgeBetween(node, start, end)) return true
+      if (state.selection.hasEdgeBetween(node, start, end)) {
+        this.updateSelection(props)
```

Comment:
Hmm yeah good point, I forgot about the main goal haha. Okay hmm, that still feels weird so I wonder if there's another way to do it.

Would it work for us to set a `data.isNative` property on the `data` inside `onSelect`, and then use that to trigger the existing native behavior logic? It seems like there's a mention of `data.isNative` inside `onSelect` already, but that it's not actually being used by the core plugin right now. I wonder if fixing that would solve this too? But without needing to mess with calling updating from inside `shouldComponentUpdate`?